src/index.ts

```diff
@@ -46,16 +46,76 @@ export default class DataSource {
       .join('&');
 
     const url = `${this.sorcererUrlBase}/${this.key}?${paramStr}`;
+    const isTokenBuilder = options['body'] && JSON.parse(options['body']).tokenApiVersion;
 
     options['cacheTime'] = options['cacheTime'] || 10 * 1000;
     options['headers'] = options['headers'] || {};
 
     options['headers']['x-reverse-proxy-ttl'] = options['cacheTime'] / 1000;
-    options['headers']['x-mi-cbe'] = CD._hashForRequest(url, options);
+    options['headers']['x-mi-cbe'] = isTokenBuilder
+      ? this.generateTokenBuilderHash(options)
+      : this.generateHash(params, options);
 
     return CD.get(url, options);
   }
 
+  /**
+   *
+   * @param options
+   */
+  generateTokenBuilderHash(options = {}): number | string {
+    options = structuredClone(options);
+    const { tokenApiVersion, tokens = [] } = JSON.parse(options['body']);
+    const cacheFragments = tokens.map(
+      (x: { skipCache: boolean; cacheOverride: string; value: string }) => {
+        if (x.skipCache) return true;
+        return x.cacheOverride || x.value;
+      }
```

Comment:
suggestions: 
- it's better to use more readable name like `token` instead of `x`
- you don't need to destructure token elements, since you access them via dot operator, so you can also just do `tokens.map((token) => { if token.skipCache ... }`
- ideally we don't want to include anything if `skipCache` is `true`, currently the array will contain the actual value `true`. Think of a way to exclude it all together or at least return an empty string instead of boolean.